### ISSUE TITLE
Multi-selecting many instances of a selecting causes slowdown and stall

### ISSUE THREAD

**Issue by manfrin** [atom/find-and-replace#692]
### Prerequisites

* [x] Can you reproduce the problem in [safe mode](http://flight-manual.atom.io/hacking-atom/sections/debugging/#check-if-the-problem-shows-up-in-safe-mode)?
* [x] Are you running the [latest version of Atom](http://flight-manual.atom.io/hacking-atom/sections/debugging/#update-to-the-latest-version)?
* [x] Did you check the [debugging guide](flight-manual.atom.io/hacking-atom/sections/debugging/)?
* [x] Did you check the [FAQs on Discuss](https://discuss.atom.io/c/faq)?
* [x] Are you reporting to the [correct repository](https://github.com/atom/atom/blob/master/CONTRIBUTING.md#atom-and-packages)?
  * I believe so -- other multi-select issues showed up as being moved to this repo.
* [x] Did you [perform a cursory search](https://github.com/issues?q=is%3Aissue+user%3Aatom+-repo%3Aatom%2Felectron) to see if your bug or enhancement is already reported?

For more information on how to write a good [bug report](https://github.com/atom/atom/blob/master/CONTRIBUTING.md#how-do-i-submit-a-good-bug-report) or [enhancement request](https://github.com/atom/atom/blob/master/CONTRIBUTING.md#how-do-i-submit-a-good-enhancement-suggestion), see the `CONTRIBUTING` guide.

### Description

Multi-select on a large file with many matches starts to crawl and then crash after trying to select many.

I had built a csv from some output and realized I formatted the newline wrong. The file wasn't super huge (64kb), so I figured I would just select all and replace instead of regenerating. I couldn't think of the select-all command off the top of my head, so I selected the first `\n` and then held down CTRL+D -- figuring it would just take 10 seconds to select through (I know -- kinda silly when a feature is built to do what I was trying to do, but I thought just selecting this way would be quicker than looking up the right way). 

However, just a few seconds in, Atom began to crawl, and then it just stopped responding. Cursor was stuck, atom didnt respond to any commands, and I had to force quit out. 

I was able to reproduce it easily, and I made a small example file that I confirmed replicates the issue. 

I'm on Atom 1.6.0, and I realize there's 1.6.1 out -- but I did not see any recently closed issues that addressed things related to this. I also could not find any other issue that referenced this (but I did find a few that might benefit from my sample file: https://github.com/atom/find-and-replace/issues/603, https://github.com/atom/find-and-replace/issues/655). 

I am on OS X El Capitain on a beefy MBP, so I don't think the available RAM/resources is at issue (although I do have vagrant running and a few servers). 

### Steps to Reproduce

1. Open provided sample file. 
2. Select something that is repeated (I was selecting `\n`).
3. Hold down CTRL+D

**Expected behavior:** Atom selects the next instance rapidly as the command is held down, until the end.

**Actual behavior:** Rapidly selected for a few seconds and then began to crawl and quickly stalled.

### Versions

1.6.0, 1.6.1, OSX El Capitain.

[sample.txt](https://github.com/atom/find-and-replace/files/198687/sample.txt)

(I had to rename the file to .txt for github upload -- issue still occurs with that extension)

**Comment by lee-dohm** [comment]
In step 3 do you mean <kbd>Cmd+D</kbd>, `find-and-replace:select-next`?

**Comment by lee-dohm** [comment]
This is also probably related to https://github.com/atom/atom/issues/10153

**Comment by manfrin** [comment]
Yep, `find-and-replace:select-next`. The related issue does look similar (I didn't know if I should file this here or at atom/atom).

**Comment by lee-dohm** [comment]
You filed in the correct place :grinning:

I just suspect that we'll need to handle key repeated commands in a general way and wanted to link the two Issues together.

**Comment by ssebbass** [comment]
Same issue, multi-line editing more than 10 lines it's almost impossible.
